src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -722,18 +743,26 @@ private void SearchPackagesAndRefreshUpdateCount(bool useCacheForUpdates)
         /// <summary>
         /// This method is called from several event handlers. So, consolidating the use of JTF.Run in this method
         /// </summary>
-        internal async Task SearchPackagesAndRefreshUpdateCountAsync(string searchText, bool useCacheForUpdates, IVsSearchCallback pSearchCallback, IVsSearchTask searchTask)
+        internal async Task SearchPackagesAndRefreshUpdateCountAsync(string searchText, bool useCachedPackages, IVsSearchCallback pSearchCallback, IVsSearchTask searchTask)
         {
             await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
+            //If we're on the Updates tab, actually load the Installed tab's data and apply a filter.
+            //ItemFilter filterToLoad = _topPanel.Filter == ItemFilter.UpdatesAvailable ? ItemFilter.Installed : _topPanel.Filter;
+            ItemFilter filterToRender = _topPanel.Filter;
+
             var loadContext = new PackageLoadContext(ActiveSources, Model.IsSolution, Model.Context);
 
-            if (useCacheForUpdates)
+            if (useCachedPackages)
             {
                 loadContext.CachedPackages = Model.CachedUpdates;
             }
+            else //Invalidate cache
+            {
+                FlagTabAsLoaded(filterToRender, isLoaded: false);
```

Comment:
Disables UI filtering for Installed/Updates tabs. The following Load will be a hard "Refresh".